libraries/chain/authorization_manager.cpp

```diff
@@ -318,12 +318,12 @@ namespace eosio { namespace chain {
       }
 
       try {
-         optional<permission_name> linked_permission = lookup_linked_permission(authorizer_account, scope, act_name);
-         if( !linked_permission )
+         std::optional<permission_name> linked_permission = lookup_linked_permission(authorizer_account, scope, act_name);
+         if( !linked_permission.has_value() )
```

Comment:
`std::optional<T>` has an overloaded `bool` operator, so there is not a need to change these lines.

Reply:
Yeah, I think I tracked down the issue with this.  The compiler was complaining about this but it appears that maybe it had to do with some implicit conversions that was causing ambiguity.